src/CommandingOfficers/CommanderAve.java

```diff
@@ -0,0 +1,534 @@
+   * outEvents will be populated with any MapChangeEvents that result.
+   */
+  private void addSnow(double amount, GameMap gameMap, GameEventQueue outEvents)
+  {
+    log("============ Adding snow ========================================");
+    for( XYCoord xyc : ownedProperties )
+    {
+      // Boost the amount of snow power this property has.
+      snowMap[xyc.xCoord][xyc.yCoord] += amount;
+      if( gameMap.getEnvironment(xyc).weatherType != Weathers.SNOW )
+      {
+        Environment newEnvi = Environment.getTile(gameMap.getEnvironment(xyc).terrainType, Weathers.SNOW);
+        GameEvent event = new MapChangeEvent(xyc, newEnvi);
+        outEvents.add(event);
+      }
+    }
+  }
+
+  /**
+   * Spread snow around.
+   * outEvents will be populated with any MapChangeEvents that result.
+   */
+  private void relevelSnow(GameMap gameMap, GameEventQueue outEvents)
+  {
+    log("  Starting snow map:");
+    log(getSnowMapAsString());
+
+    HashSet<XYCoord> roots = new HashSet<XYCoord>();
+    HashSet<XYCoord> frontier = new HashSet<XYCoord>(); // The next set of tiles to expand.
+    HashSet<XYCoord> disconnected = new HashSet<XYCoord>(); // Tiles with snow, but that haven't been expanded.
+
+    // Initialize our problem space.
+    // We start by expanding the faucets/owned props. All other snow tiles go to unused for now.
+    for( XYCoord prop : ownedProperties )
+    {
+      //sortedLeaves.add(new SnowPail(prop, snowMap[prop.xCoord][prop.yCoord]));
+      frontier.add(prop);
+    }
+    // Label all snow tiles as disconnected to begin, and relabel as needed.
+    for( int y = 0; y < gameMap.mapHeight; ++y )
+    {
+      for( int x = 0; x < gameMap.mapWidth; ++x )
+      {
+        if( snowMap[x][y] > 0 )
+        {
+          XYCoord coord = new XYCoord(x, y);
+          if( !ownedProperties.contains(coord) )
+            disconnected.add(coord);
+        }
+      }
+    }
+
+    int spreadIteration = 0;
+    while( !frontier.isEmpty() && spreadIteration < MAX_SNOW_SPREAD_RANGE )
+    {
+      spreadIteration++;
+
+      // Keep a reference so we remember where the snow wants to go.
+      // Within the while loop below, we should always read from oldSnowMap and write to snowMap.
+      int[][] oldSnowMap = getSnowMapClone();
+
+      // Prepare the next round of snow expansion.
+      PriorityQueue<SnowPail> sortedLeaves = new PriorityQueue<SnowPail>(); // The current set of tiles to expand.
+      sortedLeaves.clear();
+      for( XYCoord front : frontier )
+      {
+        sortedLeaves.add(new SnowPail(front, oldSnowMap[front.xCoord][front.yCoord]));
+      }
+      frontier.clear();
+
+      // We want to process the current batch of leaves deepest-first, so unload our queue (which
+      // is shallowest-first) into a stack and use that instead.
+      Stack<XYCoord> leafStack = new Stack<XYCoord>();
+      while(!sortedLeaves.isEmpty()) leafStack.push(sortedLeaves.poll().snowCoord);
```

Comment:
Now, I could be wrong here, but couldn't you just adjust the sorting function? No other cases of its use are immediately jumping out at me...

Reply:
A `PriorityQueue<SnowPail>` called `workingSet` is used below, and is sorted the natural (least first) way.